src/definition/scheduler/IJob.ts

```diff
@@ -0,0 +1,23 @@
+import { IUser } from '../users';
+import { IHttp, IModify, IPersistence, IRead } from './../accessors';
+import { IJobResult } from './IJobResult';
+
+export interface IJob {
+    /**
+     * The name of the job. As this property is not unique,
+     * we recommend setting this to something that is
+     * user friendly since the list of job names will be visible
+     * to the server administrator.
+     */
+    name: string;
```

Comment:
Thoughts to everyone watching this thread about requiring an `id: string` property? As right now the `name: string` is simply for displaying on the UI the list of jobs and a history of them. But if an App wants to cancel a job or view the job's history there is not a way to do it currently.

Reply:
Couldn't we use the name for that? Prepend it with the `appId` and a colon and should be good to go IMO 😛

Reply:
For our internal usage, yes. But for Apps to query, no. Because what if they want the same name for a job that runs based upon something a user does but doesn't want that detail to show up in the job history and they want to query all of the jobs done by a user? That's why I'm considering an id..

Reply:
> Because what if they want the same name for a job that runs based upon something a user does but doesn't want that detail to show up in the job history and they want to query all of the jobs done by a user?

I'm not sure I quite understand what you mean by this. Could you maybe give an example?

Reply:
Let's say you only want one job to run but you have two internal jobs and their behavior is massively based upon a configuration. The setting is changed in the admin area and when that happens you want to replace the job, that way the admin isn't aware of the different types and only sees one job in the job history and that type of stuff. Does that make sense?

Reply:
I don't understand why there would be one job running multiple internal jobs. Shouldn't that just be separate jobs then?

Reply:
Imagine a payroll or something. For auditing purposes there should only appear as one job in the log. But because payroll is an extremely complicated item, the developer wants to separate the logic out from one giant job into different ones. So internally there are several jobs by the same name that way it appears as one job. I know this is an example that is stretching it, but I do see some valid reasons for not having the name be unique.

Reply:
gotta admit I didn't follow the use case either :/

Reply:
I think I understand what @graywolf336 means.
IMO if an app wants to split one scheduled job into multiple parts, then that's something the app itself can solve and not something that should be solved by the scheduling API

Reply:
And that's why I'm think we probably should have an id for the jobs and not rely on the name 🤔